### DISCUSSION TITLE
Help with selecting from the same table twice with different filters (WITH ... AS ...)

### DISCUSSION THREAD

**Discussion by cameron1024** [diesel-rs/diesel#3309] [Q&A]
(This question was originally posted on urlo: https://users.rust-lang.org/t/how-to-implement-this-pattern-in-diesel/80714)

I'm trying to port the following SQL to diesel:
```
WITH foo_table AS ( SELECT foo_bar_id, json AS foo FROM foo_bar WHERE key = '1' ),
     bar_table AS ( SELECT foo_bar_id, json AS bar FROM foo_bar WHERE key = '2' )
SELECT foo, bar FROM foo_table INNER JOIN bar_table USING foo_bar_id
```
Approximate `schema.rs` looks like:
```rust
table! {
    foo_bar (id) {
        id -> Int8,
        key -> Numeric,
        json -> Nullable<Jsonb>,
        bytes -> Bytea,
        foo_bar_id -> Int8,
    }
}


```
TLDR: I have two different types of entity in the `foo_bar` table, and I'd like to be able to use them as "pseudo-tables" (not sure if that's the right term) that I can use in the DSL. Unfortunately, I don't have any ability to change the database schema whatsoever, otherwise I'd probably write a migration to split these into two columns.

Any advice is much appreciated, and let me know if more information is needed

**Comment by cameron1024** [comment]
Update: this question got answered on urlo. Seems like it needs the `alias` macro like:
```rust
let (foo_table, bar_table) = alias!(foo_bar as foo_table, foo_bar as bar_table);
```
